ibis/backends/flink/compiler.py

```diff
@@ -541,3 +541,33 @@ def visit_CountDistinct(self, op, *, arg, where):
         if where is not None:
             arg = self.if_(where, arg, self.f.array(arg)[2])
         return self.f.count(sge.Distinct(expressions=[arg]))
+
+    def visit_MapContains(self, op: ops.MapContains, *, arg, key):
+        return self.f.array_contains(self.f.map_keys(arg), key)
+
+    def visit_Map(self, op: ops.Map, *, keys, values):
+        if op.keys.dtype.value_type.nullable:
+            non_nullable_dtype = op.keys.dtype.value_type.copy(nullable=False)
+            keys = self.cast(keys, dt.Array(non_nullable_dtype))
+
+        return self.f.map_from_arrays(keys, values)
+
+    def visit_MapMerge(self, op: ops.MapMerge, *, left, right):
+        def get_non_nullable_keys(op, expr: sge.Expression) -> sge.Expression:
+            keys = self.f.map_keys(expr)
+            if op.dtype.key_type.nullable:
+                non_nullable_dtype = op.dtype.key_type.copy(nullable=False)
+                keys = self.cast(keys, dt.Array(non_nullable_dtype))
```

Comment:
Is `array<int not null>` actually valid flink syntax?

Reply:
I'm not following from the code why nullability matters here.

Reply:
Apparently:

1. Yes that is valid flink syntax
2. Nullability matters because it is enforced by flink